rust/connlib/tunnel/src/device_channel/tun/tun_android.rs

```diff
@@ -50,24 +52,27 @@ pub(crate) struct IfaceDevice(Arc<AsyncFd<IfaceStream>>);
 
 #[derive(Debug)]
 pub(crate) struct IfaceStream {
-    fd: RawFd,
+    fd: Closeable<RawFd>,
 }
 
 impl AsRawFd for IfaceStream {
     fn as_raw_fd(&self) -> RawFd {
-        self.fd
+        self.fd.as_raw_fd()
     }
 }
 
 impl Drop for IfaceStream {
     fn drop(&mut self) {
-        unsafe { close(self.fd) };
+        unsafe { close(self.fd.as_raw_fd()) };
     }
 }
 
 impl IfaceStream {
     fn write(&self, buf: &[u8]) -> std::io::Result<usize> {
-        match unsafe { write(self.fd, buf.as_ptr() as _, buf.len() as _) } {
+        match self
+            .fd
+            .with(|fd| unsafe { write(fd, buf.as_ptr() as _, buf.len() as _) })?
```

Comment:
It would be great to have `SAFETY` comments on here.

Reply:
I'll create an issue to add the missing `SAFETY` to all `unsafe`.

Reply:
#2277